docs/index.rst

```diff
@@ -114,6 +116,33 @@ even configuration to run the automation using `SauceLabs`_. For more
 information about what web browsers you can use, check Selenium's `WebDriver`_
 documentation.
 
+Testing With Pytest
+---------------------
+
+To run all tests::
+
+    $ pytest
+
+It is possible to run a specific subset of tests::
+
+    $ pytest test_case.py
+    $ pytest test_case.py::TestClass
+    $ pytest test_case.py::TestClass::test_case_name
+
+To get more verbose output, or run multiple tests::
+
+    $ pytest tests/ -v
+    $ pytest tests/robottelo/test_decorators.py \
+             tests/robottelo/test_cli.py
+
+To test The Foreman's API, CLI or UI, use the following commands respectively::
+
+    $ pytest tests/foreman/api/
+    $ pytest tests/foreman/cli/
+    $ pytest tests/foreman/ui/
+
+For more information about Python's `pytest`_ module, read the documentation.
+
```

Comment:
@ntkathole, hey this looks great, I think it will be a very useful addition to docs. There are some other tricks in pytest that took me personally some time to discover, please consider adding:

`pytest tests/foreman/{cli,api,ui}/test_host.py`
-- will collect from three directories in one run

`pytest tests/foreman/cli/test_host.py -k negative`
-- k option will search in testcase names, in this case it will run just negative tests

`pytest tests/foreman/cli/test_host.py -n 4`
-- n option lets you run tests in several threads, in this case 4

hope this helps